### PR TITLE
part 2 TCP routing cleanups/bug fixes for header matcher

### PR DESCRIPTION
**Pull request by rshriram**
Assorted bug fixes in ServiceEntry implementation using the 
filter chain match's destination subnets for proper handling
of CIDR ranges.
Also cleaned up Egress tests, substituting httpbin with google/bing.

And fixed bugs in multicluster code that was incorrectly assigning clusterIDs to consul/cloudfoundry/service entry registries, breaking service entry logic where a service can have multiple VIPs. (FYI @sdake / @baodongli or whoever wrote this piece of logic)

Fixes https://github.com/istio/istio/issues/6695, https://github.com/istio/istio/issues/6146, https://github.com/istio/istio/issues/5916, https://github.com/istio/istio/issues/5498, 

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -114,9 +114,6 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 		return nil, err
 	}
 
-	// ensure services are ordered to simplify generation logic
-	sort.Slice(services, func(i, j int) bool { return services[i].Hostname < services[j].Hostname })
```
Comment (GregHanson) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: is this no longer a concern?  Or do you think it should go in the aggregate registry?

Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: we sort by time later in function

**Pull request merged**